pyop2/compilation.py

```diff
@@ -498,13 +501,34 @@ def __init__(self, code, argtypes):
         raise CompilationError("Don't know what compiler to use for platform '%s'" %
                                platform)
     dll = compiler.get_so(code, extension)
+    if code.local_kernel.events:
```

Comment:
Is this check valid? If `events` is still empty we still may want to profile `solve` and `inverse`.

Reply:
Solve and inverse need to be called on some matrix and the matrix needs to be filled with data through a local assembly kernel. So the events variable would never be empty if inverse and solve are called/want to be profiled. But I will change it anyways, so that someone else does not have the same thoughts as you later.